### ISSUE TITLE
Visual Studio plugin doesn't work with System.Collctions.Immutable >1.1.37

### ISSUE THREAD

**Issue by ssrmm** [rsdn/nitra#43]
When compiling a generated plugin the [reference to `System.Collections.Immutable`](https://github.com/rsdn/nitra/blob/464d6fc76b72c4339c02da8281c6da2e1f1de8c7/Nitra.LanguageCompiler/Templates/XXLanguageFullNameXXVsPackage/Languages/NitraCommonIde.dll) is not resolved as expected: If the `Languages` subfolder contains a `System.Collections.Immutable.dll`(which it usually does), then that one takes precedence over the one from the NuGet package specified in the hint-path.

For both the Nitra and the NitraCSharp plugin this is currently version 1.1.37. Therefor the compiled plugin-dll references version 1.1.37 instead of 1.2.0 (as specified by the reference in the `.nproj`file). This by itself isn't a problem yet, but it hides one:
Visual Studio 2015 currently ships `System.Collections.Immutable` 1.1.37 as a private assembly (see `C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\PrivateAssemblies`) and an assembly redirect for all version prior to 1.1.37 (see `%LOCALAPPDATA%\Microsoft\VisualStudio\14.0\devenv.exe.config`). In order to reduce the file size of VSIXs MSBuild excludes VisualStudio assembles from it.

In order to make the reference to `System.Collections.Immutable` resolve correctly, I changed it to require the specific version:
```XML
<Reference Include="System.Collections.Immutable, Version=1.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
  <HintPath>packages\System.Collections.Immutable.1.2.0\lib\portable-net45+win8+wp8+wpa81\System.Collections.Immutable.dll</HintPath>
  <Private>True</Private>
</Reference>
```
This however breaks the plugin: MSBuild still excludes the dll from the VSIX, however 1.2.0 can no longer be supplied by VS itself and an assembly binding failure is happening at runtime. This will also happen as soon as Nitra is updated to a newer version of `System.Collections.Immutable`.

I've found [a blog](http://comealive.io/Forcing-DLLs-To-Vsix/) that provides a way to force the assembly to be included in the VSIX, however I didn't thy it out yet. The alternative would be to stay with 1.1.37 (or whatever VS ships with a future update) everywhere. This doesn't seem very useful though, because VS 2015 will probably get two more updates and that's it.

### TL;DR
1. `System.Collections.Immutable` should be referenced with specific version to ensure that the assembly is resolved as expected.
2. `System.Collections.Immutable` should be included in the VSIX __or__ it should be ensured that Nitra doesn't use a version >1.1.37 anywhere.

**Comment by VladD2** [comment]
Maybe you did not update the source code from the github repository? I can't find reference to this assembly with version > 1.1.37.

**Comment by ssrmm** [comment]
Sorry, looks like I forgot to link to it.

Yes, I meant [this one](https://github.com/rsdn/nitra/blob/464d6fc76b72c4339c02da8281c6da2e1f1de8c7/Nitra.LanguageCompiler/Templates/XXLanguageFullNameXXVsPackage/XXLanguageFullNameXXVsPackage.csproj#L129-L131).

**Comment by VladD2** [comment]
It's a bug. But in my comuter it work.

**Comment by VladD2** [comment]
I fix it. Please, test it.

**Comment by ssrmm** [comment]
It also works for me right now. I discovered this because I was doing something else wrong :smile:

Basically it works because the NuGet assembly is never referenced. If you check the references of the compiled dll in IlSpy/DotPeek, you will see that it still references 1.1.37. This is because "specific version" is not set and because `System.Collections.Immutable` in the `Languages` folder before the one in the `packages` folder. The search order for assemblies is described [here](http://stackoverflow.com/questions/1882038/hintpath-vs-referencepath-in-visual-studio/2733113#2733113).

**Comment by ssrmm** [comment]
That's the easy-fix solution. Worrying about updates to `System.Collections.Immutable` can be done when there is an actual need to do so, I guess.

According to JetBrains/Nitra#9 it is supposed to be thrown out in the long run anyway.
